### ISSUE TITLE
Any plan to upgrade requests and urllib3 version?

### ISSUE THREAD

**Issue by jenny-he** [boto/botocore#1745]
We recently got OAST critical alert with botocore due to the embedded urllib3 (CVE-2018-20060)
 and requests module(CVE-2018-18074), OAST suggested to upgrade the urllib3 > 1.24.3 and requests > 2.20. 
Is there any plan on the way to upgrade them?

**Comment by JordonPhillips** [comment]
We will get rid of it at some point. Currently we do not use the vendored requests for anything but a backwards compatibility shim on an exception class (as previously it had to be imported from the vendored requests). If you are not importing or using it then you're fine.

A while ago we added a deprecation warning that gets sent if you actually use the vendored requests. I just opened a PR to make that warning print by default: #1748 

The next step will be to remove it and add a thin shim for that previously mentioned exception class. This will be a ways off because we want to give people some time to adapt now that the warning will be shown by default. I can't give a specific timeline for that, but we will have to do it before Python 3.8 goes GA since the vendored code imports things that are being removed.
